README.md

```diff
@@ -1 +1,191 @@
+
+This project was inspired by this [issue from the Valhalla project](https://github.com/valhalla/valhalla/issues/3021).
+To summarize, Valhalla is capable of building routing tilesets and running routes on those tiles. However, the estimated
+time of arrival can be fairly inaccurate, especially in urban areas with high traffic. The estimated speeds were more
+akin to driving alone at night, which is not ideal since most people drive in the day in urbanized areas.
+
+Conflation aims to provide a *statistical approach* to estimating of driving speeds. Specifically, this project
+estimates speeds across different road classes using open-source GPS trace data. The road classes used will be taken
+from [OpenStreetMap (OSM)](https://www.openstreetmap.org/), which delineates between motorways, trunks, primary roads,
+secondary roads, residential, etc. To further refine our estimates, we will also split up our results by geographic
+area (country and region), as well as by urban and rural settings.
+
+The main idea
+
+### Method
+
+A rough list of steps this script makes is as follows:
+
+1. Create an `output/` folder where results will be stored.
+    1. This script store intermediate results on disk, so that if a run is interrupted (either intentionally or
+       accidentally) it can automatically pick up from where it left off. Since we eventually intend to have this script
+       runnable on the entire planet, not having to make repeated API calls or repeat calculations will be useful.
+    2. This script also will store the final results in this output folder (`output/results/`).
+2. Break up the given bounding box into smaller sections (to make it more manageable on the API).
+3. Make API calls to the specified GPS trace source, pulling trace data on all the bounding box sections.
+4. Filter the trace data.
+    1. The driving sequence needs to contain enough trace points.
+    2. The total distance traveled must exceed some threshold.
+    3. The total time elapsed must exceed some threshold.
+    4. The average speed needs to be within some threshold, which tells us that the user is likely driving as opposed to
+       walking, biking, riding on a train, etc.
+    5. The adjacent trace points should be within some small distance of each other (i.e. there shouldn't be any large
+       gaps in the sequence data).
+5. Perform [map matching using the Valhalla service](https://github.com/valhalla/valhalla/blob/master/docs/api/map-matching/api-reference.md).
+   This provides us a set of per-edge speed approximations. These per-edge approximations will be stored on disk using 
+   the country and region as keys. 
+6. Filter the map matching results.
+    1. The results need to have a threshold percentage of successful matches.
+7. An overall result will be derived from the per-edge measurements from Step 5, using a statistical approach.
+
+## Structure
+
+There is a specific JSON structure that this script outputs. Here is an example:
+
+```json
```

Comment:
When I specify this as JSON my linter automatically formats the arrays on different lines. If this is bothersome I can try changing it haha

Reply:
Here's one I hand generated recently. You can use it in the readme for illustrative purposes. I'm going to make another repo in the org dedicated specifically to the format and the current most recent version of the configuration. I'll probably get to that next week.

```json
[{
  "iso3166-1": "",
  "iso3166-2": "",
  "urban": {
    "way": [80,40,30,30,25,20,15,10],
    "link_exiting": [60,40,40,35,30],
    "link_turning": [60,30,25,25,20],
    "roundabout": [25,25,20,20,20,20,15,15],
    "driveway": 15,
    "alley": 10,
    "parking_aisle": 10,
    "drive-through": 10
  },
  "rural": {
    "way": [95,60,50,40,35,25,20,10],
    "link_exiting": [55,45,40,40,35],
    "link_turning": [50,35,35,30,25],
    "roundabout": [45,35,25,25,20,20,20,10],
    "driveway": 15,
    "alley": 10,
    "parking_aisle": 15,
    "drive-through": 10
  }
}]
```